Modules/permissions.py

```diff
@@ -131,7 +132,7 @@ def real_decorator(func):
     return real_decorator
 
 
-def require_channel(message: str = "This command must be invoked in a channel."):
+def require_channel(*args, **kwargs):
```

Comment:
It should still be possible to have a nice signature without varargs here. Simply have one parameter which can be either the message or the function. The same goes for `require_dm`.

Reply:
Oh, and `len(args)` can also be zero, which is a key reason i cannot have a clean signature as python wouldn't accept it.
```py
@require_channel
async def...
```
would invoke `require_channel` with zero arguments

```py
@require_channel()
```
would invoke `require_channel` with exactly one argument: the function
```py
@require_channel(message="potato")
```
would invoke require_channel` with exactly one argument and exactly one kwarg

Reply:
`require_channel(message: Union[str, Callable] = "default message")` would work fine. To go through the options:
* `@require_channel` would pass the function as the first and only argument (i.e. message).
* `@require_channel()` would use the default message.
* `@require_channel(message="message")` and `@require_channel("message")` both pass the provided message.

You would then merely need to test whether `message` is a function or a string.

Reply:
For an example, see `pytest.fixture`.